test/helpers/PermalinksTest.ts

```diff
@@ -4,27 +4,27 @@ describe('Permalinks', () => {
     describe('forRoom', () => {
         it('should generate a URL for a room ID', () => {
             const roomId = "!test:example.org";
-            const expected = `https://matrix.to/#/${roomId}`;
+            const expected = `https://matrix.to/#/${encodeURIComponent(roomId)}`;
```

Comment:
not encoding is deliberate for compatibility with some clients: the SDK should be able to parse encoded and unencoded URLs, but should not be producing encoded URLs.

Reply:
ok, that's no problem, I've changed it

> Historically, clients have not produced URIs which are fully encoded. Clients should try to interpret these cases to the best of their ability. For example, an unencoded room alias should still work within the client if possible.

> The components of the matrix.to URI (<identifier> and <extra parameter>) are to be percent-encoded as per RFC 3986

Though the spec does imply the bad links should no longer be generated when it probably should not.